src/home-page/Content/Movies/MoviesWrapper.tsx

```diff
@@ -1,23 +1,72 @@
-import { FC } from 'react';
+import { FC, useContext, useState } from 'react';
 
-import styled from 'styled-components';
+import {
+  MovieDetailsContext,
+  AddMovieModal,
+  DeleteMovie,
+  Modal,
+} from '@shared';
+import { Movie as MovieModel } from '@api/Movies';
 
 import { Movie } from './Movie';
 import { MoviesProps } from './Movies.models';
-
-const Wrapper = styled.div`
-  display: grid;
-  justify-content: space-between;
-  gap: 3rem;
-  grid-template-columns: repeat(auto-fill, 320px);
-`;
+import { Wrapper } from './Movies.styles';
+import { MenuItemsIds } from './Movies.constants';
 
 export const MoviesWrapper: FC<MoviesProps> = ({ movies }) => {
+  const { showMovieDetails } = useContext(MovieDetailsContext);
+  const [editMovieOpen, setEditMovieOpen] = useState(false);
+  const [deleteMovieOpen, setDeleteMovieOpen] = useState(false);
+  const [targetMovie, setTargetMovie] = useState<MovieModel | null>(null);
+  const showEditModal = editMovieOpen && targetMovie;
+  const showDeleteModal = deleteMovieOpen && targetMovie;
+
   return (
-    <Wrapper>
-      {movies.map((movie) => (
-        <Movie movie={movie} key={movie.id} />
-      ))}
-    </Wrapper>
+    <>
+      <Wrapper>
+        {movies.map((movie) => (
+          <Movie
+            onMenuItemClick={(id) => {
```

Comment:
I would say this chunk of js doesn't fit too nicely into JSX-block, so I would rather defined that as a `const`